File: java/po/world_simulator/entities/animals/Sheep.java
package po.world_simulator.entities.animals;

import java.util.Map;

import po.world_simulator.Vector2;
import po.world_simulator.entities.Animal;


public class Sheep extends Animal {
    public Sheep(Vector2 position, int power, int age) {
        super(power, 4, position, "Sheep", age, false);
    }

    public Sheep(Vector2 position) {
        this(position, 4, 0);
    }

    public Sheep(Map<String, String> entityData){
        this(
            new Vector2(entityData.get("position")),
            Integer.parseInt(entityData.get("power")),
            Integer.parseInt(entityData.get("age"))
        );
    }
}

File: python/po/world_simulator/entities/animals/Sheep.py
from po.world_simulator.entities.Animal import Animal
from po.world_simulator.Vector2 import Vector2

class Sheep(Animal):
    def __init__(self, position_or_data, power=4, age=0):
        if isinstance(position_or_data, dict):
            super().__init__(
                int(position_or_data.get("power", 4)),
                4,
                Vector2(position_or_data.get("position")),
                "Sheep",
                int(position_or_data.get("age", 0)),
                False
            )
        else:
            super().__init__(power, 4, position_or_data, "Sheep", age, False)